doc/main/tbb_userguide/Migration_Guide/Task_API.rst

```diff
@@ -325,3 +325,66 @@ task:
         }
         tg.wait();
     }
+
+Deferred task creation
+----------------------
+The low-level task API of TBB separates task creation from actual spawning. This separation allows to
+postpone task spawning while having parent task blocked(and final result production) from premature leave. 
+For example, ``RootTask``, ``ChildTask``, and ``CallBackTask`` are the user-side functors that
+inherit ``tbb::task`` and implement its interface. Then blocking of ``RootTask`` from premature leave 
+and waiting on it is implemented as: 
+
+.. code:: cpp
+
+    #include <tbb/task.h>
+
+    int main() {
+        // Assuming RootTask, ChildTask, and CallBackTask are defined.
+        RootTask& root = *new(tbb::task::allocate_root()) RootTask{};
+
+        ChildTask&    child    = *new(root.allocate_child()) ChildTask{/*params*/};
+        CallBackTask& cb_task  = *new(root.allocate_child()) CallBackTask{/*params*/};
+
+        root.set_ref_count(3);
+        
+        tbb::task::spawn(child);
+        
+        register_callback([cb_task&](){
+            tbb::task::enqueue(cb_task);
+        });
+
+        root.wait_for_all();
+        // Control flow will rich here only after both ChildTask and CallBackTask are executed,
```

Comment:
not clear what it reaches, maybe you meant **Control flow is reached**

Reply:
This piece of programming jargon ([Control Flow](https://en.wikipedia.org/wiki/Control_flow) ) means that CPU executing thid program will reach here only when both ChildTask and CallBackTask are executed

Reply:
Ok, I see now, thanks, then change it to reach instead of rich

Reply:
channged